server/app.ts

```diff
@@ -581,19 +623,54 @@ apiRouter.route("/checkin").post(authenticateWithReject, postParser, async (requ
 	}
 });
 
+// Endpoint for adding tags to existing groups of attendees
+apiRouter.route("/data/addTag/:tag").put(authenticateWithReject, postParser, async (request, response) => {
+	let tag: string = request.params.tag;
+	let currentTag: string = request.body.currentTag;
+	if (!currentTag) {
+		response.status(400).json({
+			"error": "Must provide existing tag"
+		});
+		return;
+	}
+	try {
+		await Attendee.update({
+			["tags." + currentTag]: {
+				$exists: true
+			},
+			["tags." + tag]: {
+				$exists: false
+			},
+		}, {
+			["tags." + tag]: {
+				checked_in: false
+			}
+		}, {
+			multi: true
+		});
+		response.status(200).json({
+			"success": true
+		});
+	}
+	catch (e) {
+		console.log(e);
+		response.status(500).json({
+			"error": "An error occurred while adding tag to attendees"
+		});
+	}
+});
+
 app.use("/api", apiRouter);
 
 const indexTemplate = Handlebars.compile(fs.readFileSync(path.join(__dirname, STATIC_ROOT, "index.html"), "utf8"));
 app.route("/").get(authenticateWithRedirect, async (request, response) => {
-	let attendees = await Attendee.find().sort({ tag: "asc" });
+	let attendees = await Attendee.find();
 	let tags: string[] = attendees.reduce((prev, current) => {
-		if (prev.indexOf(current.tag) === -1) {
-			// Escape possible HTML in tags
-			let tag: string = current.tag.replace("&", "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;");
-			prev.push(tag);
-		}
+		let tagsToAdd: string[] = Object.keys(current.tags).filter((t) => { return prev.indexOf(t) === -1; });
+		prev = prev.concat(tagsToAdd);
 		return prev;
 	}, <string[]> []);
+	tags.sort();
```

Comment:
As we learned from @bunsenmcdubbs, please add a comparator function here just so we know it's sorting the way we want